### PR TITLE
updated description for code

### PR DESCRIPTION
**Pull request by lezwon**
Missed updating the description in the previous PR. This PR updates the description for `xm.mark_step()` and `MpDeviceLoader`

### CODE REVIEW

File: API_GUIDE.md [hunk 1]
```diff
@@ -99,8 +99,8 @@ for data, target in train_loader:
 This snippet highlights how easy it is to switch your model to run on XLA. The
 model definition, dataloader, optimizer and training loop can work on any device.
 The only XLA-specific code is a couple lines that acquire the XLA device and
-step the optimizer with a <b>barrier</b>. Calling
-`xm.optimizer_step(optimizer, barrier=True)` at the end of each training
+step the optimizer. Calling
```
Comment (jysohn23) [inline API_GUIDE.md:102]: Instead of "step the optimizer", "mark the step".

File: API_GUIDE.md [hunk 2]
```diff
@@ -138,9 +138,8 @@ There are three differences between this multidevice snippet and the previous
 single device snippet:
 
 - `xmp.spawn()` creates the processes that each run an XLA device.
-- `ParallelLoader` loads the training data onto each device.
-- `xm.optimizer_step(optimizer)` no longer needs a barrier. ParallelLoader
-automatically creates an XLA barrier that evaluates the graph.
+- `MpDeviceLoader` loads the training data onto each device.
+- `xm.optimizer_step(optimizer)` creates an XLA barrier that evaluates the graph and update the model's parameters.
```
Comment (jysohn23) [inline API_GUIDE.md:142]: The description for `xm.optimizer_step()` should stay as is as it's still the `pl.MpDeviceLoader` that ends up marking the step (https://github.com/pytorch/xla/blob/master/torch_xla/distributed/parallel_loader.py#L37), unless `xm.optimizer_step` is called with `barrier` kwarg.

Comment (lezwon) [reply API_GUIDE.md:142]: @jysohn23 Since the previous [example](https://github.com/lezwon/xla/blob/code_desc_update/API_GUIDE.md#running-on-a-single-xla-device)  for running on a single xla device no longer uses `xm.optimizer_step(optimizer, barrier=True)` it wouldn't make sense to have  _`xm.optimizer_step(optimizer)` no longer needs a barrier._ in the doc. Would something like _`xm.optimizer_step(optimizer)` consolidates the gradients between cores and issues the XLA device step computation_ work?

File: API_GUIDE.md (hunk 1 above)
Comment (lezwon) [reply API_GUIDE.md:102]: will do 👍
